agent/engine/task_manager_linux.go

```diff
@@ -0,0 +1,44 @@
+// +build linux
+
+// Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package engine
+
+// SetupPlatformResources sets up platform level resources
+func (mtask *managedTask) SetupPlatformResources() error {
+	if mtask.Task.CgroupEnabled() {
```

Comment:
> We don't currently have an example outside of cgroups, though.

ENIs are, but are being handled differently right now.  We'll want to converge this in the future so that we can have something a bit more consistent.  (We're cheating with ENIs because we *do* have a Docker container involved that's necessary for Docker to share namespaces between containers.)